### PR TITLE
Applying the snake_case convention throughout the repo

### PR DESCRIPTION
**Pull request by agnivade**
- Also removing the file extension where not needed
- Adding {{ }} on a few old pages

### CODE REVIEW

**Comment by agnivade** [comment]
This PR became a bit larger than expected. Hope its not too big :stuck_out_tongue_closed_eyes: 

Xref- #478

File: pages/common/comm.md [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 - Get lines only found in first file, saving the result to a third file:
 
-`comm -23 {{file1}} {{file2}} > file1-only.txt`
+`comm -23 {{file1}} {{file2}} > file1_only.txt`
```
Comment (waldyrious) [inline pages/common/comm.md]: I wonder if "file1_only" should be wrapped in {{}} too.

File: pages/common/ed.md [hunk 2]
```diff
@@ -28,7 +28,7 @@
 
 - Write the current document to a new file (the filename can be omitted if `ed` was called with an existing file):
 
-`w {{filename.txt}}`
+`w {{file}}`
```
Comment (waldyrious) [inline pages/common/ed.md]: I'd rather keep "filename", which IMO makes it slightly clearer that this doesn't accept a path.

File: pages/common/gzip.md [hunk 3]
```diff
@@ -12,12 +12,12 @@
 
 - Compress a file specifying the output filename:
 
-`gzip -c {{file.ext}} > compressed-file.ext.gz`
+`gzip -c {{file.ext}} > compressed_file.ext.gz`
```
Comment (waldyrious) [inline pages/common/gzip.md]: Same as above: should "compressed_file" be wrapped in {{}}?

File: pages/common/haxelib.md [hunk 4]
```diff
@@ -16,4 +16,4 @@
 
 - Install the development version of a library from a Git repository:
 
-`haxelib git {{libname}} {{GIT-URL}}`
+`haxelib git {{libname}} {{GIT_URL}}`
```
Comment (waldyrious) [inline pages/common/haxelib.md]: Not sure this needs to be uppercase. I'd be fine with "git_url"

File: pages/common/license.md [hunk 5]
```diff
@@ -8,7 +8,7 @@
 
 - Create a license with custom filename:
 
-`license -o {{filename.txt}} {{license_name}}`
+`license -o {{file}} {{license_name}}`
```
Comment (waldyrious) [inline pages/common/license.md]: Same as above: if this only accepts filenames (not full paths), I believe "filename" makes that clearer.

File: pages/common/nc.md [hunk 6]
```diff
@@ -16,11 +16,11 @@
 
 - Serve a file:
 
-`cat somefile.txt | nc -l {{port}}`
+`cat some_file.txt | nc -l {{port}}`
```
Comment (waldyrious) [inline pages/common/nc.md]: {{}}?

File: pages/common/redis-cli.md [hunk 7]
```diff
@@ -20,4 +20,4 @@
 
 - Execute Redis command:
 
-`redis-cli {{redis command}}`
+`redis-cli {{redis_command}}`
```
Comment (waldyrious) [inline pages/common/redis-cli.md]: Nice catch. This is particularly troublesome in clients that make highlight tokens via color only (tldr-pages/tldr-node-client#18)

File: pages/common/unzip.md [hunk 8]
```diff
@@ -8,7 +8,7 @@
 
 - Extract zip files(s) to given path:
 
-`unzip {{files(s)}} -d {{/path/to/put/extracted/files}}`
+`unzip {{file(s)}} -d {{/path/to/put/extracted_file(s)}}`
```
Comment (waldyrious) [inline pages/common/unzip.md]: the first argument could perhaps be called "compressed file(s)"?

File: pages/linux/mdadm.md [hunk 9]
```diff
@@ -4,7 +4,7 @@
 
 - Create array:
 
-`mdadm --create {{/path/to/raid_device_file}} --level {{ raid level }} --raid-devices {{ number of disks}} {{/path/to/disk_device_file}}`
+`mdadm --create {{/path/to/raid_device_file}} --level {{ raid_level }} --raid-devices {{ number_of_disks}} {{/path/to/disk_device_file}}`
```
Comment (waldyrious) [inline pages/linux/mdadm.md]: did you deliberately not remove the spaces inside the {{}}?

**Comment by waldyrious** [comment]
Rather large indeed, but comprehensiveness rules! Great job :) I've left some inline comments.

File: pages/linux/mdadm.md (hunk 9 above)
Comment (ostera) [inline pages/linux/mdadm.md]: Looks like a simple overlook :) nothing that can't be fixed

File: pages/common/ed.md (hunk 2 above)
Comment (agnivade) [inline pages/common/ed.md]: thanks, I was in two minds about this.

File: pages/common/comm.md (hunk 1 above)
Comment (agnivade) [inline pages/common/comm.md]: Right. And I should remove the extension too.

File: pages/linux/mdadm.md (hunk 9 above)
Comment (agnivade) [inline pages/linux/mdadm.md]: oops, yes an overlook indeed. Good catch.

File: pages/common/ed.md (hunk 2 above)
Comment (waldyrious) [inline pages/common/ed.md]: we ought to codify this case in the upcoming update to CONTRIBUTING.md :)

Comment (waldyrious) [inline pages/common/ed.md]: we ought to codify this case in the upcoming update to CONTRIBUTING.md :)

File: pages/common/nc.md [hunk 10]
```diff
@@ -16,11 +16,11 @@
 
 - Serve a file:
 
-`cat somefile.txt | nc -l {{port}}`
+`cat {{some_file}} | nc -l {{port}}`
```
Comment (waldyrious) [inline pages/common/nc.md]: This change will probably conflict with #966 -- a rebase will be necessary.

Comment (agnivade) [inline pages/common/nc.md]: Yup. I know. I have it ready.

**Comment by agnivade** [comment]
Rebased with latest.

File: pages/common/nc.md [hunk 11]
```diff
@@ -16,11 +16,11 @@
 
 - Serve a file:
 
-`nc -l {{port}} < {{somefile.txt}}`
+`nc -l {{port}} < {{some_file}}`
```
Comment (waldyrious) [inline pages/common/nc.md]: Sorry for nitpicking, but isn't the "some" here redundant? Especially considering the other pages, which use a bare "file" or "filename".

**Comment by waldyrious** [comment]
LGTM apart from the last inline comment I added. With that addressed, I'd OK the merging of this PR :)

File: pages/common/nc.md (hunk 11 above)
Comment (agnivade) [inline pages/common/nc.md]: Sure. No worries. :)

**Pull request merged**